3_optimize_ws_goroutines/kqueue_darwin.go

```diff
@@ -0,0 +1,101 @@
+// +build darwin
+
+package main
+
+import (
+	"log"
+	"reflect"
+	"sync"
+	"syscall"
+
+	"github.com/gorilla/websocket"
+)
+
+type epoll struct {
+	fd          int
+	connections map[int]*websocket.Conn
+	changes     []syscall.Kevent_t
+	lock        *sync.RWMutex
+}
+
+func MkEventsCollector() (*epoll, error) {
+	fd, err := syscall.Kqueue()
+	if err != nil {
+		return nil, err
+	}
+	_, err = syscall.Kevent(fd, []syscall.Kevent_t{{
```

Comment:
You can write it shorter this way:
`if _, err = syscall.Kevent....; err != nil { .. }`

Reply:
Sounds good!